src/main/java/de/tum/in/www1/artemis/service/TutorParticipationService.java

```diff
@@ -0,0 +1,129 @@
+package de.tum.in.www1.artemis.service;
+
+import de.tum.in.www1.artemis.domain.ExampleSubmission;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.TutorParticipation;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.TutorParticipationStatus;
+import de.tum.in.www1.artemis.repository.ExampleSubmissionRepository;
+import de.tum.in.www1.artemis.repository.TutorParticipationRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service Implementation for managing TutorParticipation.
+ */
+@Service
+@Transactional
+public class TutorParticipationService {
+
+    private final Logger log = LoggerFactory.getLogger(TutorParticipationService.class);
+    private final ExampleSubmissionRepository exampleSubmissionRepository;
+
+    @Value("${server.url}")
+    private String ARTEMIS_BASE_URL;
+
+    private final TutorParticipationRepository tutorParticipationRepository;
+
+    public TutorParticipationService(TutorParticipationRepository tutorParticipationRepository,
+                                     ExampleSubmissionRepository exampleSubmissionRepository) {
+        this.tutorParticipationRepository = tutorParticipationRepository;
+        this.exampleSubmissionRepository = exampleSubmissionRepository;
+    }
+
+    /**
+     * Save a tutorParticipations.
+     *
+     * @param tutorParticipation the entity to save
+     * @return the persisted entity
+     */
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public TutorParticipation save(TutorParticipation tutorParticipation) {
+        log.debug("Request to save TutorParticipation : {}", tutorParticipation);
+        return tutorParticipationRepository.saveAndFlush(tutorParticipation);
+    }
+
+    /**
+     * Get all the tutorParticipations.
+     *
+     * @return the list of entities
+     */
+    @Transactional(readOnly = true)
+    public List<TutorParticipation> findAll() {
+        log.debug("Request to get all TutorParticipations");
+        return tutorParticipationRepository.findAll();
+    }
+
+    /**
+     * Get all the tutorParticipations.
+     *
+     * @param pageable the pagination information
+     * @return the list of entities
+     */
+    @Transactional(readOnly = true)
+    public Page<TutorParticipation> findAll(Pageable pageable) {
```

Comment:
Please get rid of this method. We won't need this. We are always only interested in tutor participations in the context of one exercise (or course), we are never interested in really all of them (even if it would be pages).

Reply:
Done